File: visible-ios/visible/visible/NodeActionsMenu.swift
import SwiftUI

/// The Edit details / Rename / Move / Delete actions shown for a node, both from
/// the current node's top-bar overflow `Menu` and from a child card's
/// `.contextMenu`. Move and Delete are omitted when `isRoot` is true — the root
/// house has no parent, so it can be neither re-parented nor deleted.
struct NodeActionsMenu: View {
    let onEdit: () -> Void
    let onRename: () -> Void
    let onMove: () -> Void
    let onDelete: () -> Void
    let isRoot: Bool

    var body: some View {
        Button("Edit details", action: onEdit)
        Button("Rename", action: onRename)
        if !isRoot {
            Button("Move", action: onMove)
            Button("Delete", role: .destructive, action: onDelete)
        }
    }
}

File: visible-android/app/src/main/java/fm/bae/visible/ui/NodeActionsMenu.kt
package fm.bae.visible.ui

import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable

/**
 * The Edit details / Rename / Delete menu shown for a node, both from the current
 * node's overflow button and from a child card's long-press. Delete is omitted
 * when [canDelete] is false (the root house has no parent and can't be deleted).
 * Each action dismisses the menu before running.
 */
@Composable
fun NodeActionsMenu(
    expanded: Boolean,
    onDismiss: () -> Unit,
    onEdit: () -> Unit,
    onRename: () -> Unit,
    onDelete: () -> Unit,
    canDelete: Boolean,
) {
    DropdownMenu(expanded = expanded, onDismissRequest = onDismiss) {
        DropdownMenuItem(
            text = { Text("Edit details") },
            onClick = {
                onDismiss()
                onEdit()
            },
        )
        DropdownMenuItem(
            text = { Text("Rename") },
            onClick = {
                onDismiss()
                onRename()
            },
        )
        if (canDelete) {
            DropdownMenuItem(
                text = { Text("Delete") },
                onClick = {
                    onDismiss()
                    onDelete()
                },
            )
        }
    }
}
